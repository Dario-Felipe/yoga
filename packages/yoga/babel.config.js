module.exports = {
  extends: '../../babel.config.js',
  plugins: ['transform-import-css'],
};
