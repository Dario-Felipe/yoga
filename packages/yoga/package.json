{
  "name": "@gympass/yoga",
  "version": "5.2.0-alpha.4",
  "description": "Gympass component library",
  "main": "./src",
  "sideEffects": false,
  "scripts": {
    "build": "yarn build:web && yarn build:native && yarn build:esm",
    "build:web": "NODE_ENV=web babel ./src --out-dir dist/cjs",
    "build:esm": "NODE_ENV=esm babel ./src --out-dir dist/esm",
    "build:native": "NODE_ENV=native babel ./src --out-dir dist/cjs",
    "prebuild": "rm -rf ./dist",
    "postbuild": "cp -r ./src/Fonts ./dist/fonts",
    "prepublishOnly": "node ../../scripts/prepublish.js --rn"
  },
  "keywords": [
    "Gympass",
    "components",
    "styled-components",
    "react",
    "design-system"
  ],
  "homepage": "https://github.com/Gympass/yoga#readme",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "directory": "dist"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Gympass/yoga.git"
  },
  "bugs": {
    "url": "https://github.com/Gympass/yoga/issues"
  },
  "dependencies": {
    "@gympass/yoga-common": "^0.1.2-alpha.4",
    "@gympass/yoga-helpers": "^0.1.1-alpha.0",
    "@gympass/yoga-icons": "^0.11.2-alpha.5",
    "@gympass/yoga-tokens": "^1.1.0-alpha.0",
    "@ptomasroos/react-native-multi-slider": "^2.2.2",
    "downshift": "^6.1.0",
    "prop-types": "^15.7.2",
    "proxy-polyfill": "^0.3.2",
    "rc-slider": "^9.7.1",
    "react-google-font-loader": "^1.1.0"
  },
  "devDependencies": {
    "@testing-library/react": "^11.2.5",
    "@testing-library/react-native": "^7.2.0",
    "babel-plugin-inline-react-svg": "^2.0.1",
    "styled-components": "^5.2.1"
  },
  "peerDependencies": {
    "styled-components": "^5.2.1"
  }
}
