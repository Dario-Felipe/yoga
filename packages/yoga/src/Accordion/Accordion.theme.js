const Accordion = ({ spacing, fontSizes, colors }) => ({
  padding: {
    standard: spacing.medium,
    large: spacing.large,
  },
  paddingArrow: {
    total: spacing.xxsmall,
  },
  gap: {
    header: spacing.xxxsmall,
  },
  margin: {
    top: spacing.xxxsmall,
  },
  fontSize: {
    itemList: fontSizes.small,
    marker: fontSizes.xxsmall,
  },
  lineHeight: {
    itemList: fontSizes.large,
  },
  colors: {
    content: colors.text.secundary,
  },
});

export default Accordion;
