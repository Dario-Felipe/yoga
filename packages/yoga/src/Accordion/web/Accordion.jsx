import React, { useState } from 'react';
import styled, { css } from 'styled-components';
import { string, node, bool } from 'prop-types';
import { ChevronDown } from '@gympass/yoga-icons';
import { Text } from '@gympass/yoga';
import Divider from '../../Divider';

const Accordion = ({ title, subtitle, children, disabled }) => {
  const [open, setOpen] = useState(false);

  const AccordionWrapper = styled.div`
    border: none;
    display: flex;
    flex-direction: column;
    position: relative;
    width: 100%;
    z-index: 1;

    ${({
      theme: {
        yoga: {
          colors: { white, elements },
        },
      },
    }) => {
      return `
      background: ${disabled ? elements.backgroundAndDisabled : white}
    `;
    }}

    hr {
      bottom: 0;
      left: 0;
      margin: 0;
      position: absolute;
      z-index: -1;
    }
  `;

  const Header = styled.button`
    align-items: center;
    background-color: transparent;
    border: none;
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    width: 100%;

    ${({
      theme: {
        yoga: {
          components: { accordion },
        },
      },
    }) => {
      return `
      padding: ${
        subtitle
          ? `${accordion.padding.standard}px`
          : `${accordion.padding.large}px ${accordion.padding.standard}px`
      };
    `;
    }}

    &:focus-visible {
      ${({
        theme: {
          yoga: {
            colors: { elements },
          },
        },
      }) => {
        return `
        background: ${elements.backgroundAndDisabled};

      `;
      }}
    }

    ${disabled &&
    css`
      cursor: not-allowed;
    `}
  `;

  const Title = styled.div`
    align-items: flex-start;
    display: flex;
    flex-direction: column;

    ${({
      theme: {
        yoga: {
          components: { accordion },
        },
      },
    }) => {
      return `
      gap: ${subtitle ? accordion.gap.header : 0}px;
      margin: ${subtitle ? 0 : `${accordion.paddingArrow.total}px 0`};
    `;
    }}

    ${disabled &&
    css`
      ${({
        theme: {
          yoga: {
            colors: { text },
          },
        },
      }) => {
        return `
          color: ${text.secundary};
        `;
      }}
    `}
  `;

  const AccordionContent = styled.div`
    height: auto;
    max-height: ${({ isOpen }) => (isOpen ? '9999px' : '0')};
    overflow: hidden;
    transition: max-height 200ms ease-in-out;
  `;

  const ArrowWrapper = styled.div`
    align-items: center;
    display: flex;
    justify-content: center;

    svg {
      transform: ${({ isOpen }) =>
        isOpen ? 'rotate(180deg)' : 'rotate(0deg)'};
      transition: all 200ms ease-out 0s;
    }

    ${({
      theme: {
        yoga: {
          components: { accordion },
        },
      },
    }) => {
      return `
      padding: ${accordion.paddingArrow.total}px;
    `;
    }}
  `;

  return (
    <AccordionWrapper>
      <Header
        disabled={disabled}
        onClick={() => {
          setOpen(!open);
        }}
      >
        <Title>
          <Text color={disabled && 'deep'}>{title}</Text>

          <Text.Small color="deep">{subtitle}</Text.Small>
        </Title>

        <ArrowWrapper isOpen={open}>
          <ChevronDown
            fill={disabled ? '#6B6B78' : '#D8385E'}
            width={24}
            height={24}
          />
        </ArrowWrapper>
      </Header>

      <AccordionContent isOpen={open}>{children}</AccordionContent>
      <Divider />
    </AccordionWrapper>
  );
};

Accordion.propTypes = {
  title: string.isRequired,
  subtitle: string,
  children: node.isRequired,
  disabled: bool,
};

Accordion.defaultProps = {
  subtitle: undefined,
  disabled: false,
};

export default Accordion;
