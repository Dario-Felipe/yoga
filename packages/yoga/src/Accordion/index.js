import Accordion from './web';

export default Accordion;
