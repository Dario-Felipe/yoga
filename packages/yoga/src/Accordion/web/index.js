import Accordion from './Accordion';
import Content from './Content';

Accordion.Content = Content;

export default Accordion;
