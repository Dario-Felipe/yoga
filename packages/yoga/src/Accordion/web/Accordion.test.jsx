import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Text, Button } from '@gympass/yoga';
import { ThemeProvider } from '../..';

import Accordion from '.';

describe('<Accordion />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { baseElement } = render(
      <ThemeProvider>
        <Accordion title="Title">
          <Accordion.Content>
            <Text>Content</Text>

            <Button small inverted>
              Small button
            </Button>
          </Accordion.Content>
        </Accordion>
      </ThemeProvider>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
