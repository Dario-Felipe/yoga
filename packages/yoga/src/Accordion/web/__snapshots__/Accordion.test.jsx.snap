// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should match snapshot 1`] = `
.c9 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

.c8 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 32px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #FFFFFF;
  color: #D8385E;
}

.c8 svg {
  width: 16px;
  height: 16px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c8:not([disabled]):hover,
.c8:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c8:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c8:active svg {
  fill: #FFFFFF;
}

.c8 svg {
  fill: #D8385E;
}

.c8:active {
  background-color: rgba(255,255,255,0.75);
  color: rgba(216,56,94,0.75);
}

.c8:active svg {
  fill: rgba(216,56,94,0.75);
}

.c8:not([disabled]):hover,
.c8:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(255,255,255,0.45);
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
}

.c0 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
  background: #FFFFFF;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 24px 20px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0px;
  margin: 8px 0;
}

.c6 {
  height: auto;
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
}

.c7 {
  padding: 0 20px 20px;
}

<body>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        <div
          class="c2"
        >
          <p
            class="c3"
          >
            Title
          </p>
          <p
            class="c4"
            color="deep"
          />
        </div>
        <div
          class="c5"
        >
          <svg
            fill="#D8385E"
            height="24"
            width="24"
          />
        </div>
      </button>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <p
            class="c3"
          >
            Content
          </p>
          <button
            class="c8"
          >
            Small button
          </button>
        </div>
      </div>
      <hr
        class="c9"
      />
    </div>
  </div>
</body>
`;
